File: C++/best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size() == 1 or prices.size() == 0) return 0;
        
        unsigned int profit = 0;
        unsigned int low_price = prices[0];
        
        for(int i = 0; i < prices.size(); i++) {
            if(prices[i] < low_price) low_price = prices[i];
            if(prices[i] - low_price > profit) profit = prices[i] - low_price;
        }
        return profit;
    }
};

File: Python3/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 1 or len(prices) == 0: return 0
        
        profit = 0
        low_price = prices[0]
        
        for price in prices:
            if price < low_price: low_price = price
            if price - low_price > profit: profit = price - low_price
        
        return profit
